### PR TITLE
Support changing power state

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

**Comment by stbenjam** [comment]
@dtantsur Would you mind taking a look at this one?

File: ironic/resource_ironic_node_v1.go [hunk 1]
```diff
@@ -145,6 +148,25 @@ func resourceNodeV1() *schema.Resource {
...
 			},
+			"power_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"target_power_state": {
+				Type:     schema.TypeString,
+				Optional: true,
+
+				// If power_state is same as target_power_state, we have no changes to apply
```
Comment (dtantsur) [inline ironic/resource_ironic_node_v1.go]: This is only triggered when target_power_state is changed, right?

File: ironic/resource_ironic_node_v1.go [hunk 2]
```diff
@@ -145,6 +148,25 @@ func resourceNodeV1() *schema.Resource {
...
+				DiffSuppressFunc: func(_, old, new string, d *schema.ResourceData) bool {
+					return new == d.Get("power_state").(string)
+				},
+			},
+			"power_state_timeout": {
+				Type:         schema.TypeString,
```
Comment (dtantsur) [inline ironic/resource_ironic_node_v1.go]: Why a string? Is it some terraform limitation?

File: ironic/resource_ironic_node_v1.go [hunk 3]
```diff
@@ -628,3 +666,40 @@ func (workflow *provisionStateWorkflow) changeProvisionState(target nodes.Target
 
 	return false, nodes.ChangeProvisionState(workflow.client, workflow.d.Id(), *opts).ExtractErr()
 }
+
+// Call Ironic's API and change the power state of the node
+func changePowerState(client *gophercloud.ServiceClient, d *schema.ResourceData, target nodes.TargetPowerState) error {
+	timeout, err := strconv.Atoi(d.Get("power_state_timeout").(string))
+	if err != nil {
+		timeout = 300
```
Comment (dtantsur) [inline ironic/resource_ironic_node_v1.go]: Looks like you're silently ignoring errors here, is it a good idea?

File: ironic/resource_ironic_node_v1.go (hunk 1 above)
Comment (stbenjam) [reply ironic/resource_ironic_node_v1.go]: Yup

File: ironic/resource_ironic_node_v1.go (hunk 2 above)
Comment (stbenjam) [reply ironic/resource_ironic_node_v1.go]: Terraform's type system is extremely limited... the next version is better, but 0.11.3 does have TypeInt, I've fixed it.

File: ironic/resource_ironic_node_v1.go (hunk 3 above)
Comment (stbenjam) [reply ironic/resource_ironic_node_v1.go]: Fixed

Comment (stbenjam) [reply ironic/resource_ironic_node_v1.go]: I've updated this code to leave this unspecified if the user didn't give us a timeout.  

I still needed some value to wait for how long to poll Ironic for target_provision_state to clear, so there's still a default value here.  There's no other way to know if the request is finished, right?

**Comment by stbenjam** [comment]
@dtantsur Would you mind taking a look at this one again?

**Pull request merged**